cinder/volume/drivers/vmware/volumeops.py

```diff
@@ -984,6 +984,33 @@ def get_summary(self, datastore):
                                         self._session.vim, datastore,
                                         'summary')
 
+    def get_datastore_custom_attributes(self, datastore):
+        retrieve_fields = self._session.invoke_api(vim_util,
+                                                   'get_object_property',
+                                                   self._session.vim,
+                                                   datastore,
+                                                   'availableField')
+        if retrieve_fields:
+            custom_fields = {}
+            for field in retrieve_fields:
+                for v in field[1]:
+                    custom_fields[v.key] = v.name
+
+            retrieve_result = self._session.invoke_api(vim_util,
+                                                       'get_object_property',
+                                                       self._session.vim,
+                                                       datastore,
+                                                       'customValue')
+            if retrieve_result:
+                custom_attributes = {}
+                for val in retrieve_result:
+                    for i in val[1]:
```

Comment:
Again, why do we need a `[1]`? What's in `[0]`? Are there more? Why can we ignore them?

Reply:
removed